stixcore/products/product.py

```diff
@@ -315,3 +315,199 @@ def __repr__(self):
     @classmethod
     def from_packets(cls, packets):
         raise NotImplementedError
+
+
+class GenericProduct(BaseProduct):
+    def __init__(self, *, service_type, service_subtype, ssid, control, data,
+                 idb_versions=defaultdict(SCETimeRange), **kwargs):
+        """
+        Generic product composed of control and data
+
+        Parameters
+        ----------
+        control : stix_parser.products.quicklook.Control
+            Table containing control information
```

Comment:
this is nice that some of the methods are now move to this super class.

what is your definition difference between Base and Generic Product?

Reply:
Yea more work can be done here, the idea was the `Base` is inherited by other products but `GenericProduct` is not?